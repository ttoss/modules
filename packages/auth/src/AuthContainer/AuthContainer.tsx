import { color, Flex, FlexProps } from '@ttoss/ui';

export type AuthContainerProps = FlexProps & { backgroundImageUrl?: string };

export const AuthContainer = ({
  sx,
  backgroundImageUrl,
  ...props
}: AuthContainerProps) => {
  return (
    <Flex
      {...props}
      sx={{
        height: '100vh',
        justifyContent: 'center',
        alignItems: 'center',
        margin: 0,
        backgroundImage: (t: any) =>
          backgroundImageUrl
            ? `url("${backgroundImageUrl}")`
            : `linear-gradient(
            to right top,
            ${color.alpha('primary', 0.5)(t)},
            ${color.alpha('secondary', 0.5)(t)}
          )`,
        backgroundPosition: 'center',
        backgroundRepeat: 'no-repeat',
        backgroundSize: 'cover',
        ...sx,
      }}
    />
  );
};
