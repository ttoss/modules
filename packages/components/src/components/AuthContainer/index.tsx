import * as React from 'react';
import { Flex, Image, Link, Text } from 'theme-ui';

import LogoDemo from '../../components/LogoDemo/index';

type LinkButtonProps = {
  children: React.ReactNode | React.ReactNodeArray;
  href: string;
};

const LinkButton = ({ children, href }: LinkButtonProps) => {
  return (
    <Link
      sx={{
        marginTop: '24px',
        textDecoration: 'underline',
        cursor: 'pointer',
      }}
      href={href}
    >
      {children}
    </Link>
  );
};

type AuthContainerProps = {
  children: React.ReactNode | React.ReactNodeArray;
  urlLogo?: string;
  title: string;
  links?: {
    label: string;
    href: string;
  }[];
};

const AuthContainer = ({
  children,
  urlLogo,
  title,
  links,
}: AuthContainerProps) => {
  return (
    <Flex
      sx={{
        maxWidth: '400px',
        width: '100%',
        border: 'solid 2px #444',
        padding: '32px',
        flexDirection: 'column',
        alignItems: 'center',
        color: 'text',
        fontFamily: 'Arial',
      }}
    >
      {urlLogo ? (
        <Image sx={{ maxHeight: '120px', marginY: '14px' }} src={urlLogo} />
      ) : (
        <LogoDemo />
      )}
      <Text
        sx={{
          marginY: '24px',
          textAlign: 'center',
          width: '100%',
          fontWeight: 'bold',
          textTransform: 'uppercase',
        }}
      >
        {title}
      </Text>
      {children}
      {links?.map((link, idx) => (
        <LinkButton key={idx} href={link.href}>
          {link.label}
        </LinkButton>
      ))}
    </Flex>
  );
};

export default AuthContainer;
