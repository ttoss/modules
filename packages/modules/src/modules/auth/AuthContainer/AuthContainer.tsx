import * as React from 'react';
import { Button, Flex, Image, Text } from 'theme-ui';

import LogoDemo from '../../../components/LogoDemo/index';

type LinkButtonProps = {
  children: React.ReactNode | React.ReactNodeArray;
  onClick: () => void;
};

const LinkButton = ({ children, onClick }: LinkButtonProps) => {
  return (
    <Button
      as="a"
      sx={{ color: 'black', cursor: 'pointer', marginTop: '24px' }}
      onClick={onClick}
    >
      {children}
    </Button>
  );
};

type AuthContainerProps = {
  children: React.ReactNode | React.ReactNodeArray;
  urlLogo?: string;
  title: string;
  links?: Array<{
    label: string;
    onClick: () => void;
  }>;
};

const AuthContainer = ({
  children,
  urlLogo,
  title,
  links,
}: AuthContainerProps) => {
  return (
    <Flex
      sx={{
        maxWidth: '400px',
        width: '100%',
        border: 'solid 2px #444',
        padding: '32px',
        flexDirection: 'column',
        alignItems: 'center',
        color: 'text',
        fontFamily: 'Arial',
      }}
    >
      {urlLogo ? (
        <Image sx={{ maxHeight: '120px', marginY: '14px' }} src={urlLogo} />
      ) : (
        <LogoDemo />
      )}
      <Text
        sx={{
          marginY: '24px',
          textAlign: 'center',
          width: '100%',
          fontWeight: 'bold',
          textTransform: 'uppercase',
        }}
      >
        {title}
      </Text>
      {children}
      {links?.map((link) => (
        <LinkButton key={link.label} onClick={link.onClick}>
          {link.label}
        </LinkButton>
      ))}
    </Flex>
  );
};

export default AuthContainer;
