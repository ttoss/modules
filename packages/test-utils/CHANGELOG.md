# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.4.0](https://github.com/ttoss/modules/compare/v1.3.0...v1.4.0) (2022-02-03)

### Features

- notifications module ([#36](https://github.com/ttoss/modules/issues/36)) ([374b528](https://github.com/ttoss/modules/commit/374b528e832bf7caee6fb742a8be7fdafbcbd297))

# [1.3.0](https://github.com/ttoss/modules/compare/v1.2.0...v1.3.0) (2022-01-27)

### Features

- auth logo ([#32](https://github.com/ttoss/modules/issues/32)) ([17bdf97](https://github.com/ttoss/modules/commit/17bdf978cc5f10c72452611c85778d6d75c3aa87))

# [1.2.0](https://github.com/ttoss/modules/compare/v1.1.0...v1.2.0) (2022-01-26)

**Note:** Version bump only for package @ttoss/test-utils

# [1.1.0](https://github.com/ttoss/modules/compare/v0.9.2...v1.1.0) (2022-01-24)

**Note:** Version bump only for package @ttoss/test-utils

## [0.9.2](https://github.com/ttoss/modules/compare/v0.9.1...v0.9.2) (2022-01-19)

**Note:** Version bump only for package @ttoss/test-utils

## [0.9.1](https://github.com/ttoss/modules/compare/v0.9.0...v0.9.1) (2022-01-18)

### Bug Fixes

- theme provider ([#23](https://github.com/ttoss/modules/issues/23)) ([7b5d754](https://github.com/ttoss/modules/commit/7b5d7542a7126954c55d67e7fb91c681a09e0996))

# [0.9.0](https://github.com/ttoss/modules/compare/v0.8.5...v0.9.0) (2022-01-18)

**Note:** Version bump only for package @ttoss/test-utils

## [0.8.5](https://github.com/ttoss/modules/compare/v0.8.4...v0.8.5) (2022-01-17)

**Note:** Version bump only for package @ttoss/test-utils

## [0.8.4](https://github.com/ttoss/modules/compare/v0.8.3...v0.8.4) (2022-01-17)

**Note:** Version bump only for package @ttoss/test-utils

## [0.8.3](https://github.com/ttoss/modules/compare/v0.8.2...v0.8.3) (2022-01-04)

**Note:** Version bump only for package @ttoss/test-utils

## [0.8.2](https://github.com/ttoss/modules/compare/v0.8.1...v0.8.2) (2022-01-04)

**Note:** Version bump only for package @ttoss/test-utils

## [0.8.1](https://github.com/ttoss/modules/compare/v0.8.0...v0.8.1) (2021-12-23)

**Note:** Version bump only for package @ttoss/test-utils

# [0.8.0](https://github.com/ttoss/modules/compare/v0.7.0...v0.8.0) (2021-12-23)

**Note:** Version bump only for package @ttoss/test-utils

# [0.7.0](https://github.com/ttoss/modules/compare/v0.6.2...v0.7.0) (2021-09-08)

### Features

- add useGeocoder ([d740c38](https://github.com/ttoss/modules/commit/d740c38d38f38f724888fc544dfac09f86816efb))

## [0.6.2](https://github.com/ttoss/modules/compare/v0.6.1...v0.6.2) (2021-09-07)

### Bug Fixes

- dependencies and exports ([cb8a4c4](https://github.com/ttoss/modules/commit/cb8a4c40baacaf59c90a0468b0795aaa81c87935))

## [0.6.1](https://github.com/ttoss/modules/compare/v0.6.0...v0.6.1) (2021-09-07)

**Note:** Version bump only for package @ttoss/test-utils

# [0.6.0](https://github.com/ttoss/modules/compare/v0.5.1...v0.6.0) (2021-09-07)

### Features

- maps first implementation ([3191e60](https://github.com/ttoss/modules/commit/3191e609011d764586bb0e32664dc4d110413002))

## [0.5.1](https://github.com/ttoss/modules/compare/v0.5.0...v0.5.1) (2021-09-07)

**Note:** Version bump only for package @ttoss/test-utils

# 0.5.0 (2021-09-07)

### Features

- add example to test-utils ([9d81c8e](https://github.com/ttoss/modules/commit/9d81c8ec8dfa3d6a23e645f9b26c1fcc95650417))
- add test-utils ([ad0d69e](https://github.com/ttoss/modules/commit/ad0d69e07acc82a2671389ba7691cffe846bd603))
- auth first implementation ([2309b75](https://github.com/ttoss/modules/commit/2309b7552b8e659254f334069999f04eaf3f57b0))
- creating ttoss Storybook ([a652a2e](https://github.com/ttoss/modules/commit/a652a2ee473fba647a9dd32c1a90a35936a74107))
- ui first implementation ([44ec22d](https://github.com/ttoss/modules/commit/44ec22dff6a76e25f5bef0cde3735115460b972f))
