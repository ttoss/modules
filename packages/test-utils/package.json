{
  "name": "@ttoss/test-utils",
  "version": "1.11.5",
  "description": "This package provides a number of utilities and re-exports for testing using Jest, React Testing Library, and Storybook.",
  "keywords": [
    "Jest",
    "React",
    "React Testing Library",
    "Storybook"
  ],
  "author": "ttoss",
  "contributors": [
    {
      "email": "pedro@arantespp.com",
      "name": "Pedro Arantes",
      "url": "https://arantespp.com/contact"
    }
  ],
  "publishConfig": {
    "access": "public"
  },
  "main": "dist/index.js",
  "typings": "dist/index.d.ts",
  "module": "dist/index.mjs",
  "files": [
    "dist",
    "src",
    "storybook.d.ts",
    "storybook.js"
  ],
  "license": "UNLICENSED",
  "scripts": {
    "build": "tsup",
    "postbuild": "./scripts/post-build.sh",
    "test": "jest"
  },
  "dependencies": {
    "@emotion/jest": "^11.7.1",
    "@storybook/addon-storyshots": "^6.4.14",
    "@storybook/testing-react": "^1.2.3",
    "@testing-library/dom": "^8.11.3",
    "@testing-library/jest-dom": "^5.16.1",
    "@testing-library/react": "^12.1.2",
    "@testing-library/react-hooks": "^7.0.2",
    "@testing-library/user-event": "^13.5.0",
    "faker": "5.5.3"
  },
  "peerDependencies": {
    "@storybook/client-api": ">=6.0.0",
    "@storybook/react": ">=6.0.0",
    "react": ">=16.8.0",
    "react-dom": ">=16.8.0"
  },
  "devDependencies": {
    "@ttoss/config": "^1.11.5",
    "@types/faker": "^5.5.9"
  },
  "gitHead": "2a911f70d465d6bffdd6589a62d17701f1026f01"
}
