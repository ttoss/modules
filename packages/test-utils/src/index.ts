import { createSerializer, matchers } from '@emotion/jest';
import '@testing-library/jest-dom';
import { render, RenderOptions } from '@testing-library/react';
import * as React from 'react';

/**
 * Export all the matchers for Jest to avoid the error:
 * > Property 'toHaveStyleRule' does not exist on type 'JestMatchers<HTMLElement>'.
 */
export * from '@emotion/jest';

export {
  renderHook,
  /**
   * DEPRECATED: use `act` instead.
   */
  act as renderHookAct,
} from '@testing-library/react-hooks';
export { default as userEvent } from '@testing-library/user-event';

/**
 * Add the custom matchers provided by '@emotion/jest'
 * https://emotion.sh/docs/@emotion/jest#custom-matchers
 */
expect.extend(matchers);

/**
 * Output the actual styles being applied.
 * https://emotion.sh/docs/testing
 */
expect.addSnapshotSerializer(createSerializer());

/**
 * Custom render options.
 */
let options_: RenderOptions = {};

export type { RenderOptions };

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { ...options_, ...options });

// eslint-disable-next-line import/export
export * from '@testing-library/react';

// eslint-disable-next-line import/export
export { customRender as render };

export const setOptions = (options: RenderOptions) => {
  options_ = options;
};

/**
 * Storybook
 */
export { default as initStoryshots } from '@storybook/addon-storyshots';
/**
 * This package needs `@storybook/client-api` package.
 */
export {
  composeStories,
  setGlobalConfig as setStorybookGlobalConfig,
  composeStory,
} from '@storybook/testing-react';
