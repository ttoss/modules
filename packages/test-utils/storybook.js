module.exports = require('./dist/storybook');
