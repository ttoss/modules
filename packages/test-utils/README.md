# @ttoss/test-utils

This package provides a number of utilities and re-exports for testing using [Jest](https://jestjs.io/), [React Testing Library](https://testing-library.com/docs/react-testing-library/intro), and [Storybook](https://storybook.js.org/).

## How to Configure Your Project

You can configure your project following the instructions in the [Jest](https://jestjs.io/) documentation. Basically, you need to add jest to your project, `yarn add --dev jest` and run `npx jest --init`, which will create a configuration file in your project root. If you're working with React, you'll need to change the property `testEnvironment` to `jsdom`.

```ts
// jest.config.ts
export default {
  // ...
  testEnvironment: 'jsdom',
};
```

If you're working with TypeScript, you'll also need to install [Babel](https://jestjs.io/docs/getting-started#using-babel) and create a `babel.config.js` to handle the transpilation of the ES6 code, TypeScript, and JSX.

```sh
yarn add --dev babel-jest @babel/core @babel/preset-env @babel/preset-react @babel/preset-typescript
```

```js
// babel.config.js
module.exports = {
  presets: [
    ['@babel/preset-env', { targets: { node: 'current' } }],
    /**
     * Adding this line allows you to use JSX in your files without using
     * `import React as 'react'`.
     */
    ['@babel/preset-react', { runtime: 'automatic' }],
    '@babel/preset-typescript',
  ],
};
```

## Installing the Package

```sh
yarn add --dev @ttoss/test-utils
```

## Using the Package

<!-- _You can see an example of the package on this [GitHub repository](https)._ -->

### React

This package re-exports the following libraries:

- [@testing-library/jest-dom](https://github.com/testing-library/jest-dom): custom [Jest](https://jestjs.io/) matchers to test the state of the DOM.
- Everything from [@testing-library/react](https://testing-library.com/docs/react-testing-library/intro/): APIs for working with React components.
- `renderHook` from [@testing-library/render-hooks](https://react-hooks-testing-library.com/): a utility for testing custom hooks.
- `userEvent` from [@testing-library/user-event](https://testing-library.com/docs/ecosystem-user-event/): fire events the same way the user does.

#### render

`render` is a method that allows you to render a React component with provided options. Before using it, you need to setup your testing options.

```tsx
// jest.setup.ts
import { setOptions } from '@ttoss/test-utils';

import AllProviders from './paht/to/AllProviders';

/**
 * Add global wrapper to React Testing Library `customRender`.
 */
setOptions({ wrapper: AllProviders });
```

Add `jest.setup.ts` to your `jest.config.js` file.

```ts
export default {
  // ...
  setupFilesAfterEnv: ['./jest.setup.ts'],
};
```

Finally, you write your tests like this:

```tsx
import { render, screen, userEvent } from '@ttoss/test-utils';

import Component from './Component';

test('test with render', () => {
  render(<Component />);

  userEvent.click(screen.getByText('Increment'));

  expect(screen.getByText(1)).toBeInTheDocument();
});
```

### Storybook

You can use your Storybook stories in your unit tests or create Storyshoots testing.

#### Storybook Stories

You can reuse the Storybook stories that you've already created in your unit tests. To do so, this packages uses the package [@storybook/testing-react](https://github.com/storybookjs/testing-react), that you can use this way:

```tsx
import { customRender, screen, userEvent } from '@ttoss/test-utils';
import { composeStories } from '@ttoss/test-utils/storybook';

import * as stories from './my.stories';

const { Example } = composeStories(stories);

test('check if Storybook Example story is working', () => {
  customRender(<Example />);

  expect(screen.getByText('oi')).toBeInTheDocument();
  expect(screen.getByText(0)).toBeInTheDocument();
  expect(screen.getByText('Increment')).toBeInTheDocument();
  expect(screen.getByText('StorybookDecorator')).toBeInTheDocument();
  expect(screen.getByText('JestSetupProvider')).toBeInTheDocument();

  userEvent.click(screen.getByText('Increment'));

  expect(screen.getByText(1)).toBeInTheDocument();
});
```

If your Storybook configuration has global decorators/parameters/etc and you want to use them in your tests, you can use the `setGlobalConfig` function to pass them to your stories. In your `jest.setup.ts` file, you can use the `setGlobalConfig` this way:

```tsx
// jest.setup.ts
import { setGlobalConfig } from '@ttoss/test-utils/storybook';

/**
 * Add global config to Storybook.
 * https://storybook.js.org/addons/@storybook/testing-react
 */
import * as globalStorybookConfig from './.storybook/preview';

setGlobalConfig(globalStorybookConfig);
```

#### Storyshoots
