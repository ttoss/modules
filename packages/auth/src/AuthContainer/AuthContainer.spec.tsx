import { render } from '@ttoss/test-utils';

import { AuthContainer } from './AuthContainer';

test('default background', () => {
  const { container } = render(<AuthContainer />);

  expect(container).toMatchSnapshot();
});

test('image background', () => {
  const { container } = render(
    <AuthContainer backgroundImageUrl="https://source.unsplash.com/random" />
  );

  expect(container).toMatchSnapshot();
});
