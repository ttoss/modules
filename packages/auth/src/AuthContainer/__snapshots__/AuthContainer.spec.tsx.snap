// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default background 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100vh;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0;
  background-image: linear-gradient(
              to right top,
              rgba(0,0,0,0.5),
              rgba(255,255,255,0.5)
            );
  -webkit-background-position: center;
  background-position: center;
  background-repeat: no-repeat;
  -webkit-background-size: cover;
  background-size: cover;
}

<div>
  <div
    class="emotion-0"
  />
  <div
    class="Toastify"
  />
</div>
`;

exports[`default background on mobile 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100vh;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0;
  background-image: linear-gradient(
              to right top,
              rgba(0,0,0,0.5),
              rgba(255,255,255,0.5)
            );
  -webkit-background-position: center;
  background-position: center;
  background-repeat: no-repeat;
  -webkit-background-size: cover;
  background-size: cover;
}

<div>
  <div
    class="emotion-0"
  />
  <div
    class="Toastify"
  />
</div>
`;

exports[`image background 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100vh;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0;
  background-image: url("https://source.unsplash.com/random");
  -webkit-background-position: center;
  background-position: center;
  background-repeat: no-repeat;
  -webkit-background-size: cover;
  background-size: cover;
}

<div>
  <div
    class="emotion-0"
  />
  <div
    class="Toastify"
  />
</div>
`;
